ff/src/fields/models/fp/mod.rs

```diff
@@ -64,6 +64,10 @@ pub trait FpConfig<const N: usize>: Send + Sync + 'static + Sized {
     /// which works for every modulus.
     const SQRT_PRECOMP: Option<SqrtPrecomputation<Fp<Self, N>>>;
 
+    /// Precomputed lookup table for values 0..2^16 in Montgomery form.
+    /// Otherwise, conversion to Montgomery form requires a multiplication by R^2.
+    const SMALL_ELEMENT_MONTGOMERY_PRECOMP: [Fp<Self, N>; PRECOMP_TABLE_SIZE];
```

Comment:
Have you considered doing this with `lazy_static!()` or `include_bytes!()` + `build.rs`?